src/AttributeEngine/AttributeResolver/AttributeResolver.php

```diff
@@ -74,15 +78,34 @@ public function isAttributeSupported(string $attributeId)
      *
      * @param string $attributeId
      * @param $value
-     * @return AttributeInterface
+     * @return Attribute
      */
-    public function getAttributeFor(string $attributeId, $value)
+    public function getAttributeFor(string $attributeId, $rawValue = null, AttributeValue $value = null): Attribute
```

Comment:
I'm not sure I understand why the signature here could not be:

```suggestion
    public function getAttributeFor(string $attributeId, $value = null): Attribute
```

Then within the method we could condition on `$value instanceof AttributeValue` to determine how to handle it. It feels like keeping the signature intuitive would be preferable, as this is a method that non-core developers may use in custom contexts or actions for example.

Perhaps I'm missing a piece of the puzzle here?

Reply:
Absolutely - done.

Reply:
Great :) I see you made changes relating to attribute constructors regarding this, did you also intend to do the same for `getAttributeFor`?